### PR TITLE
Only run docker-disk-gc on low-disk

### PR DESCRIPTION
**Pull request by lox**
This riffs on #325 and supersedes #317. It adds a disk free check to the cron script that cleans up docker things so that older images/containers/volumes are only purged if disk or inodes are low. 

It also increases the frequency of this to hourly (cron.hourly). I was almost tempted to run this in the `environment` hook too, so that issues are surfaced during the build if possible.

Feels @patrobinson?

### CODE REVIEW

**Comment by lox** [comment]
Yup, agreed @patrobinson. I've made those changes.

**Comment by lox** [comment]
The other thing this does is marks the instance as unhealthy if it has low disk and an image purge won't fix it.

File: packer/conf/docker/cron.hourly/docker-network-gc [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+set -euo pipefail
+
+DOCKER_PRUNE_UNTIL=${DOCKER_PRUNE_UNTIL:-4h}
+
+## ------------------------------------------
+## Prune stuff that doesn't affect cache hits
+
+docker network prune -af --filter "until=${DOCKER_PRUNE_UNTIL}"
+docker container prune --all --force --filter "until=${DOCKER_PRUNE_UNTIL}"
+docker volume prune --all --force --filter "until=${DOCKER_PRUNE_UNTIL}"
```
Comment (patrobinson) [inline packer/conf/docker/cron.hourly/docker-network-gc]: The name of this file is no longer relevant

**Comment by patrobinson** [comment]
Sounds pretty sweet. I like it! 👍

**Pull request merged**